### DISCUSSION TITLE
Buffer queues with disk cache for web hook/http server in EMQX Open Source

### DISCUSSION THREAD

**Discussion by gildevall** [emqx/emqx#11899] [Q&A]
In the docs the following section can be [found](https://www.emqx.io/docs/en/latest/getting-started/new-features.html#overload-protection-limiter-and-buffer-queue-for-bridges): 
`A generic buffer queue is also added for all bridges, allowing buffer messages generated under stressful conditions. This buffer can be configured to store messages in either memory or disk cache when external resources are unavailable, such as during network fluctuations or service downtime. Buffered messages will be sent once the service is restored. However, requests in the buffer may expire, which is a big difference compared to version 4. If the amount of buffered data exceeds the limit, it will be discarded following the First In First Out (FIFO) rule.`

I'm particularly interested in this part: `This buffer can be configured to store messages in either memory or disk cache when external resources are unavailable, such as during network fluctuations or service downtime.`

I have not been able to find any settings related to **disk cache** in relation to web hook/http server in EMQX Open Source. Is this something that is on the way or not planned? At least not for the open source version.

**Comment by savonarola** [comment]
Hello!

The buffer type is a hidden option of bridges because using a memory cache is strongly recommended.
Also, it is available through the config only.

To experiment with the feature, you may set up a required bridge and then add the following option to the config:
```
...
bridges {
  ...
  mqtt { # or any other type
      ...
      $name {
         ...
         resource_opts {
            ...
            buffer_mode = volatile_offload
         }
      }
  }
}
...
```
